vendor/src/github.com/docker/distribution/contrib/docker-integration/nginx/registry.conf

```diff
@@ -0,0 +1,231 @@
+  }
+}
+
+server {
+  listen 5442;
+  listen 5443;
+  server_name localhost;
+  ssl on;
+  ssl_certificate /etc/nginx/ssl/registry-ca+client-cert.pem;
+  ssl_certificate_key /etc/nginx/ssl/registry-ca+client-cert-key.pem;
+  ssl_client_certificate /etc/nginx/ssl/registry-ca+client-ca.pem;
+  ssl_verify_client on;
+  client_max_body_size 0;
+  chunked_transfer_encoding on;
+  location /v2/ {
+    include               docker-registry-v2.conf;
+  }
+  location / {
+    include               docker-registry.conf;
+  }
+}
+
+server {
+  listen 5444;
+  server_name localhost;
+  ssl on;
+  ssl_certificate /etc/nginx/ssl/registry-ca+client-cert.pem;
+  ssl_certificate_key /etc/nginx/ssl/registry-ca+client-cert-key.pem;
+  ssl_client_certificate /etc/nginx/ssl/registry-ca+client-ca.pem;
+  ssl_verify_client on;
+  client_max_body_size 0;
+  chunked_transfer_encoding on;
+  location /v2/ {
+    auth_basic "registry.localhost";
+    auth_basic_user_file test.passwd;
+    add_header 'Docker-Distribution-Api-Version' 'registry/2.0' always;
+    include               docker-registry-v2.conf;
+  }
+  location / {
+    auth_basic "registry.localhost";
+    auth_basic_user_file test.passwd;
+    include               docker-registry.conf;
+  }
+}
+
+server {
+  listen 5445;
+  server_name localhost;
+  ssl on;
+  ssl_certificate /etc/nginx/ssl/registry-noca-cert.pem;
+  ssl_certificate_key /etc/nginx/ssl/registry-noca-cert-key.pem;
+  client_max_body_size 0;
+  chunked_transfer_encoding on;
+  location /v2/ {
+    include               docker-registry-v2.conf;
+  }
+  location / {
+    include               docker-registry.conf;
+  }
+}
+
+server {
+  listen 5446;
+  server_name localhost;
+  ssl on;
+  ssl_certificate /etc/nginx/ssl/registry-noca-cert.pem;
+  ssl_certificate_key /etc/nginx/ssl/registry-noca-cert-key.pem;
+  client_max_body_size 0;
+  chunked_transfer_encoding on;
+  location /v2/ {
+    auth_basic "registry.localhost";
+    auth_basic_user_file test.passwd;
+    add_header 'Docker-Distribution-Api-Version' 'registry/2.0' always;
+    include               docker-registry-v2.conf;
+  }
+  location / {
+    auth_basic "registry.localhost";
+    auth_basic_user_file test.passwd;
+    include               docker-registry.conf;
+  }
+}
+
+server {
+  listen 5447;
+  server_name localhost;
+  ssl on;
+  ssl_certificate /etc/nginx/ssl/registry-noca-cert.pem;
+  ssl_certificate_key /etc/nginx/ssl/registry-noca-cert-key.pem;
+  ssl_client_certificate /etc/nginx/ssl/registry-ca+client-ca.pem;
+  ssl_verify_client on;
+  client_max_body_size 0;
+  chunked_transfer_encoding on;
+  location /v2/ {
+    include               docker-registry-v2.conf;
+  }
+  location / {
+    include               docker-registry.conf;
+  }
+}
+
+server {
+  listen 5448;
+  server_name localhost;
+  ssl on;
+  ssl_certificate /etc/nginx/ssl/registry-ca+client-cert.pem;
+  ssl_certificate_key /etc/nginx/ssl/registry-ca+client-cert-key.pem;
+  ssl_protocols       SSLv3;
```

Comment:
We should probably strip out the contrib directory. This particular line you are referencing is part of a test case to ensure Docker fails to fallback to SSLv3.  So this is something you want, just not here.